src/pyvesync/vesync.py

```diff
@@ -72,22 +73,84 @@ def energy_update_interval(self, new_energy_update):
         if new_energy_update > 0:
             self._energy_update_interval = new_energy_update
 
+    @staticmethod
+    def device_search(device, new_list):
+        """Tests if device should be removed - False = Remove"""
+        if isinstance(new_list, list) and device.cid:
+            for item in new_list:
+                device_found = False
+                if 'cid' in item:
+                    if device.cid == item['cid']:
+                        device_found = True
+                        break
+                else:
+                    logger.error('No cid found in - ' + item)
+            if not device_found:
+                logger.debug("Device removed - {} - {}".format(
+                    device.device_name, device.device_type))
+                return False
+            else:
+                return True
+
+    def new_devices(self, new_dev):
```

Comment:
same as the other one. maybe something like `should_add_device` or something else?

Reply:
I agree, renamed to `add_dev_test` for consistency (the sun was coming up when I finished this off and I forgot to revisit the names the next day)